corgidrp/l4_to_tda.py

```diff
@@ -137,6 +137,516 @@ def determine_color_cor(input_dataset, ref_star, source_star):
     return color_dataset
 
 
+def convert_spec_to_flux(input_dataset, fluxcal_factor, slit_transmission=None):
+    """
+    Flux calibrate 1-D spectroscopy spectra stored in the L4 SPEC extension.
+    The function applies COL_COR when present, propagates calibration
+    uncertainties, and records whether slit correction was applied. Requires
+    the input dataset to have already been core-throughput corrected
+    (ie SPEC header contains CTCOR=True).
+
+    Args:
+        input_dataset (corgidrp.data.Dataset): L4 dataset containing SPEC,
+            SPEC_ERR, SPEC_DQ, SPEC_WAVE, and SPEC_WAVE_ERR extensions.
+        fluxcal_factor (corgidrp.data.FluxcalFactor): absolute flux calibration
+            product used to scale the spectrum.
+        slit_transmission (tuple or list of tuples, optional): slit throughput
+            information from spec.slit_transmission(). Provide either a single
+            (slit_map, slit_x, slit_y) tuple applied to every frame or a list
+            containing one tuple per frame in input_dataset.
+
+    Returns:
+        corgidrp.data.Dataset: copy of the input dataset with the
+        SPEC/SPEC_ERR data converted to erg/(s*cm^2*Å) and
+        headers/history updated.
+        
+    """
+    if not isinstance(fluxcal_factor, FluxcalFactor):
+        raise TypeError("fluxcal_factor must be a corgidrp.data.FluxcalFactor instance.")
+
+    spec_dataset = input_dataset.copy()
+
+    # Normalize slit transmission input to per-frame list
+    if slit_transmission is None:
+        slit_per_frame = [None] * len(spec_dataset)
+    elif isinstance(slit_transmission, tuple):
+        if len(slit_transmission) != 3:
+            raise TypeError("slit_transmission tuples must be (slit_map, slit_x, slit_y).")
+        slit_per_frame = [slit_transmission] * len(spec_dataset)
+    elif isinstance(slit_transmission, list):
+        if len(slit_transmission) not in (1, len(spec_dataset)):
+            raise ValueError("slit_transmission must have length 1 or match the dataset length.")
+        slit_per_frame = list(slit_transmission)
+        for tup in slit_per_frame:
+            if not (isinstance(tup, tuple) and len(tup) == 3):
+                raise TypeError("Each slit_transmission entry must be a (slit_map, slit_x, slit_y) tuple.")
+        if len(slit_per_frame) == 1 and len(spec_dataset) > 1:
+            slit_per_frame = slit_per_frame * len(spec_dataset)
+    else:
+        raise TypeError("slit_transmission must be None, a (slit_map, slit_x, slit_y) tuple, or a list of such tuples.")
+
+    history_messages = []
+
+    for idx, frame in enumerate(spec_dataset):
+        if 'SPEC' not in frame.hdu_list:
+            raise ValueError("Input dataset does not contain a 'SPEC' extension.")
+
+        mask_loc_x = frame.ext_hdr.get('MASKLOCX') # using MASKLOCX to check if the image is coronagraphic
```

Comment:
I don't think we write this in spec mode. You may want to use `FSMLOS = 1` or just skip this check altogether for now.